### PR TITLE
Added minimal scraper

### PR DESCRIPTION
**Pull request by gthomdev**
Contains basic reddit scraper using PRAW to grab links from subreddit posts and validate them

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,9 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+.idea/inspectionProfiles/Project_Default.xml
```
Comment (davidweatherall) [inline .gitignore]: Best to just gitignore the whole /idea folder?

File: RedditBot.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        client_secret=os.environ.get('REDDIT_CLIENT_SECRET'),
+        username=os.environ.get('REDDIT_USERNAME'),
+        password=os.environ.get('REDDIT_PASSWORD')
+    )
+    with open("Comments.txt", 'w', encoding="utf-8") as infile:
+        for submission in reddit.subreddit(target_subreddit).new(limit=target_submission_count):
+            if hasattr(submission, "selftext_html"):
+                soup = BeautifulSoup(submission.selftext_html, 'html.parser')
+                for link in soup.findAll('a', href=True):
+                    if RedditBotHelpers.is_target_link_type(link.attrs['href']):
```
Comment (davidweatherall) [inline RedditBot.py]: Best to break after successfully finding a link? We don't want multiple profiles from one post

File: tests.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import unittest
+import RedditBotHelpers
+
+
+class LinkValidatorTests(unittest.TestCase):
+    def setUp(self):
+        self.u_gg_link = r'https://na.op.gg/summoners/na/JeffLynne'
+        self.op_gg_link = r'https://na.op.gg/summoners/na/JeffLynne'
+        self.blitz_gg_link = r'https://blitz.gg/lol/profile/euw1/JeffLynne'
```
Comment (davidweatherall) [inline tests.py]: Need a bunch more test cases

File: .gitignore (hunk 1 above)
Comment (gthomdev) [reply .gitignore]: Done

File: RedditBot.py (hunk 2 above)
Comment (gthomdev) [reply RedditBot.py]: Done

File: tests.py (hunk 3 above)
Comment (gthomdev) [reply tests.py]: Done

**Pull request merged**